python/ray/train/_internal/session.py

```diff
@@ -222,15 +222,14 @@ def reset(
         self.training_started = False
         self._first_report = True
 
-        # Change the working directory to the local trial directory.
-        # -> All workers on the same node share a working directory.
-        os.makedirs(storage.trial_local_path, exist_ok=True)
```

Comment:
I am actually a bit worried that we changed the working dir into the ray global session dir in the performance perspective:

For example, users may save large checkpoints into `/tmp` folder, and use up the disk storage. Intuitively we should only save the small metadata and logs in this Ray session dir.

A mitigation is to change ray global session dir onto another disk (e.g. mounted SSD with larger storage and higher bandwidth), all other logs (ray core, data, ...) will be moved together in consequence. We probably don't want our ray train featuer affects other libraries' behavior.

Reply:
`/tmp/ray` is the place where large ray objects get spilled, and I am also a bit worried that random things written by to the working directory will cause out of disk on the `/tmp` drive.

For now, it's possible to work around this by setting a different output directory for saving checkpoints, so let's just keep it as is. The other driver files are very small, so it's fine for the general case.

In the future, if there are user issues related to this, we can consider adding an environment variable to allow users to set the ray train staging path without needing to move all of the ray logs.

Reply:
Yes it'd be better to allow users to specify a custom ray train staging path. Let's create an issue/add a todo?